packages/devp2p/src/protocol/eth.ts

```diff
@@ -47,29 +53,33 @@ export class ETH extends Protocol {
   static eth65 = { name: 'eth', version: 65, length: 17, constructor: ETH }
   static eth66 = { name: 'eth', version: 66, length: 17, constructor: ETH }
 
+  
+
   _handleMessage(code: ETH.MESSAGE_CODES, data: any) {
     const payload = arrToBufArr(RLP.decode(bufArrToArr(data)))
     const messageName = this.getMsgPrefix(code)
-    const debugMsg = `Received ${messageName} message from ${this._peer._socket.remoteAddress}:${this._peer._socket.remotePort}`
+    const debugMsg = this.DEBUG ?  `Received ${messageName} message from ${this._peer._socket.remoteAddress}:${this._peer._socket.remotePort}` : undefined;
 
-    if (code !== ETH.MESSAGE_CODES.STATUS) {
+    if (code !== ETH.MESSAGE_CODES.STATUS && this.DEBUG) {
       const logData = formatLogData(data.toString('hex'), this._verbose)
       this.debug(messageName, `${debugMsg}: ${logData}`)
     }
     switch (code) {
       case ETH.MESSAGE_CODES.STATUS: {
-        assertEq(
-          this._peerStatus,
-          null,
-          'Uncontrolled status message',
-          this.debug.bind(this),
-          'STATUS'
-        )
-        this._peerStatus = payload as ETH.StatusMsg
-        const peerStatusMsg = `${
-          this._peerStatus !== undefined ? this._getStatusString(this._peerStatus) : ''
-        }`
-        this.debug(messageName, `${debugMsg}: ${peerStatusMsg}`)
+        if (this.DEBUG) {
```

Comment:
This should move below line 77 below as we need the `this._peerStatus...` stuff regardless of whether we're doing debug logging or not.